src/backend/api/meals.js

```diff
@@ -1,15 +1,97 @@
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
-  }
+// Returns all meals
+router.get("/", async(request, response) => {
+    try {
+        // knex syntax for selecting things. Look up the documentation for knex for further info
+        const titles = await knex("meals");
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
 });
-
-module.exports = router;
+//Adds a new meal
+router.post("/", async (request, response) => {
+    try {
+          const newmeals = await knex("meals").insert(request.body);
+      response.send("New meal added");
+    } catch (error) {
+      throw error;
+    }
+  });
+  //Returns meal by id
+  router.get("/:id", async (request, response) => {
+    try {
+      const meals = await knex("meals").where("id", parseInt(request.params.id));
+      response.json(meals);
+    } catch (error) {
+      throw error;
+    }
+  });
+  //Updates the meal by id	
+  router.put("/:id", async (request, response) => {
+    try {
+        const meals = await knex("meals")
+        .where("id", parseInt(request.params.id))
+        .update(request.body);
+      response.json(meals);
+    } catch (error) {
+      throw error;
+    }
+  });
+//Deletes the meal by id
+router.delete("/:id", async (request, response) => {
+    try {
+      const deleteMeal = await knex("meals")
+        .where("id", parseInt(request.params.id))
+        .del();
+      response.json(deleteMeal);
+    } catch (error) {
+      throw error;
+    }
+  });
+  router.get("/", async (req, res) => {
+    try {
+      const maxPrice = req.query.maxPrice;
+      const availableReservations = req.query.availableReservations;
+      const title = req.query.title;
+      const createdAfter = req.query.createdAfter;
+      const limit = req.query.limit;
+      // Get meals that has a price smaller than maxPrice
+      if (maxPrice) {
+        const result = await knex("meals").where("price", "<", maxPrice);
+        res.send(result);
+      }// Get meals that still has available reservations
+      else if (availableReservations) {
+        const result = await knex("meals").where("number_of_guests", ">", 0);
+        res.send(result);
+      }
+      // Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+      else if (title) {
+        const result = await knex("meals").where("title", "like", `%${title}%`);
+        res.send(result);
+      }
+      // Get meals that has been created after the date
+      else if (createdAfter) {
+        const newDate = new Date(createdAfter);
+        const result = await knex("meals").where("when", ">", newDate);
+        res.send(result);
+      }
+      // Only specific number of meals
+      else if (limit) {
+        const result = await knex("meals").limit(limit);
+        res.send(result);
+      }
```

Comment:
Good work in using knex to query the database and filter the meals directly instead of loading them and using the filter method.
One thing though that will not work, is multiple query parameters For instance, if I wanted to see meals with a certain title AND also created after a date, I will only get results based on my title parameter!